src/Microsoft.Identity.Web/ClaimsPrincipalExtensions.cs

```diff
@@ -16,7 +16,7 @@ public static class ClaimsPrincipalExtensions
         /// </summary>
         /// <param name="claimsPrincipal">Claims principal.</param>
         /// <returns>A string corresponding to an account identifier as defined in <see cref="Microsoft.Identity.Client.AccountId.Identifier"/>.</returns>
-        public static string GetMsalAccountId(this ClaimsPrincipal claimsPrincipal)
+        public static string? GetMsalAccountId(this ClaimsPrincipal claimsPrincipal)
         {
             if (claimsPrincipal == null)
```

Comment:
From a code perspective, we still need to make null checks and program defensively, because NULLs can still be passed everywhere.